adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -338,6 +338,9 @@ private Bundle getAuthTokenFromAccountManager(final AuthenticationRequest reques
                 Logger.e(TAG, AUTHENTICATOR_CANCELS_REQUEST, "", ADALError.AUTH_FAILED_CANCELLED, e);
             } catch (AuthenticatorException e) {
                 Logger.e(TAG, AUTHENTICATOR_CANCELS_REQUEST, "", ADALError.BROKER_AUTHENTICATOR_NOT_RESPONDING);
+                if (!StringExtensions.isNullOrBlank(e.getMessage()) && e.getMessage().contains(ADALError.DEVICE_CONNECTION_IS_NOT_AVAILABLE.toString())) {
```

Comment:
Also this won't work with the new bound service. Why can't we check network connection right before we talk to broker? in the case of new service, we can check connection when service is on bound.